client/statement-store/src/lib.rs

```diff
@@ -762,7 +767,35 @@ impl StatementStore for Store {
 	/// Return the decrypted data of all known statements whose decryption key is identified as
 	/// `dest`. The key must be available to the client.
 	fn posted_clear(&self, match_all_topics: &[Topic], dest: [u8; 32]) -> Result<Vec<Vec<u8>>> {
-		self.collect_statements(Some(dest), match_all_topics, |statement| statement.into_data())
+		self.collect_statements(Some(dest), match_all_topics, |statement| {
+			if let (Some(key), Some(_)) = (statement.decryption_key(), statement.data()) {
+				let public = UncheckedFrom::unchecked_from(key);
+				let mut out = None;
+				if let Err(e) = self.keystore.with_ed25519_key(
+					sp_core::crypto::key_types::STATEMENT,
+					&public,
+					&mut |pair| match statement.decrypt_private(pair) {
+						Ok(r) => out = r,
+						Err(e) => log::debug!(
+							target: LOG_TARGET,
+							"Decryption error: {:?}, for statement {:?}",
+							e,
+							HexDisplay::from(&statement.hash())
+						),
+					},
+				) {
```

Comment:
We can have with_ed25519_key use a function that return an error as param, and potentially have posted clear report this in its list for results.